File: Project3_Python/Plant.py
from Organism import Organism

import random


class Plant(Organism):
    __THE_FLOOR_OF_SPREAD = 10

    def __init__(self, world, position, strength=0):
        super().__init__(world, position, strength)
        self._is_animal = False

    def action(self):
        probability_of_spread = random.randint(0, 100)

        if probability_of_spread < self.__THE_FLOOR_OF_SPREAD:
            new_position = self.get_new_position()
            if new_position.x == -1:
                return False
            if self._world.check_if_not_occupied(new_position):
                self._world.add_organism(self.type_of_organism, new_position)
                self._world.add_comment(str(self) + " spread.")
                return True

        return False

    def get_image(self):
        file_name = "Images/" + str(self) + ".jpg"
        return file_name

File: Project2_Java/src/pl/pg/oop/project2/organisms/Plant.java
package pl.pg.oop.project2.organisms;

import javax.swing.*;
import java.awt.*;
import java.util.Random;

public abstract class Plant extends Organism {
    protected static final int THE_FLOOR_OF_SPREAD = 10;
    protected Plant(World world, Point position) {
        super(world, position);
        IsAnimal = false;
    }

    public boolean action() {
        Random random = new Random();
        int ProbabilityOfSpread = random.nextInt(100);
        if(ProbabilityOfSpread < THE_FLOOR_OF_SPREAD) {
            Point newPosition = getNewPosition();
            if(newPosition.x == -1) return false;
            if(world.checkIfNotOccupied(newPosition)) {
                world.AddOrganism(TypeOfOrganism, newPosition);
                world.addComment(this + " spread.");
                return true;
            }
        }
        return false;
    }

    @Override
    public ImageIcon getImage() {
        String FileName = "Images/"+ this +".jpg";
        return new ImageIcon(FileName);
    }
}
